packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -425,6 +425,15 @@ Future<void> _writeAndroidPluginRegistrant(FlutterProject project, List<Plugin>
       templateContent = _androidPluginRegistryTemplateNewEmbedding;
     break;
     case '1':
+      for (Map<String, dynamic> plugin in androidPlugins) {
+        if (plugin['usesEmbedding2']) {
```

Comment:
I think this will get populated based on https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/platform_plugins.dart#L76 as soon as the new embedding appears right? We probably want to check whether it has the v2 plugin and only has the v2 plugin here.

Reply:
Good point. I think these two can't be decoupled. I updated the PR to now return a set of the supported embeddings. PTAL